prog/setup_grafana.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class Prog::SetupGrafana < Prog::Base
+  subject_is :sshable
+
+  def self.assemble(sshable_id, grafana_domain:, certificate_owner_email:)
+    if grafana_domain.strip.empty? || certificate_owner_email.strip.empty?
+      fail "grafana_domain and certificate_owner_email should not be empty"
+    end
+    unless (sshable = Sshable[sshable_id])
+      fail "Sshable does not exist"
+    end
+    Strand.create(prog: "SetupGrafana", label: "start", stack: [{subject_id: sshable.id, domain: grafana_domain.strip, cert_email: certificate_owner_email.strip}])
+  end
+
+  def domain
+    frame["domain"]
+  end
+
+  def cert_email
+    frame["cert_email"]
+  end
+
+  label def start
+    bud Prog::BootstrapRhizome, {"target_folder" => "host", "subject_id" => sshable.id, "user" => sshable.unix_user}
+    hop_wait_for_rhizome
+  end
+
+  label def wait_for_rhizome
+    reap
+    hop_install_grafana if leaf?
+    donate
+  end
+
+  label def install_grafana
+    case sshable.d_check("install_grafana")
+    when "Succeeded"
+      sshable.d_clean("install_grafana")
+      pop "grafana was setup"
+    when "NotStarted"
+      sshable.d_run("install_grafana", "sudo", "host/bin/setup-grafana", domain, cert_email)
+      nap 10
+    when "InProgress"
+      nap 10
+    when "Failed"
+      Clog.emit("Install grafana failed")
+      nap 65536
+    else
+      nap 65536
```

Comment:
My understanding is we shouldn't hit this case.

```suggestion
      Clog.emit("Unexpected install_grafana result")
      nap 65536
```

Maybe we should assume a different result is the same as failure, and eliminate the extra conditional?

@fdr is this something we should create a Page for?

Reply:
I knew that this behavior is not optimal and I looked at the lifecycle of this strand as something that would be run very infrequently and someone would wait and check the result when creating it. That's why I didn't took the extra steps to create pages for it.

Reply:
In that case, maybe a page is not necessary.  However, I think it's still better to combine the Failed and unexpected cases.  I'd move the `Clog.emit` to the `else` branch, and remove the `when "Failed"` branch.